src/drivers/driver.ts

```diff
@@ -614,10 +615,64 @@ export abstract class CMakeDriver implements vscode.Disposable {
     await this.doPreCleanConfigure();
     this.configRunning = false;
 
-    return this.configure(extra_args, consumer);
+    return this.configure(trigger, extra_args, consumer);
   }
 
-  async configure(extra_args: string[], consumer?: proc.OutputConsumer, withoutCmakeSettings:boolean = false): Promise<number> {
+  async testCompilerVersion(program: string, cwd: string, arg: string | undefined, regexp: RegExp): Promise<string> {
+    const args = [];
+    if (arg) {
+      args.push(arg);
+    }
+    const child = this.executeCommand(program, args, undefined, {silent: true, cwd});
+    try {
+      const result = await child.result;
+      log.debug(localize('command.version.test.return.code', 'Command version test return code {0}', nullableValueToString(result.retc)));
+      // cl.exe version information is printed on stderr when no source files are given
+      const versionLine = (program === "cl") ? result.stderr : result.stdout; //.split("\\n")[0];
+      const match = regexp.exec(versionLine);
+      return match ? match[1] : "error";
+    } catch (e) {
+      const e2: NodeJS.ErrnoException = e;
+      log.debug(localize('compiler.version.return.code', 'Compiler version test return code {0}', nullableValueToString(e2.code)));
+      if (e2.code === 'error') {
+        return e2.code;
+      }
+      throw e;
+    }
+  }
+
+  async getCompilerVersion(compilerPath: string) : Promise<string | undefined> {
+    const compilerName = path.parse(compilerPath).name;
+    const compilerDir = path.parse(compilerPath).dir;
+    let versionSwitch;
+    let versionRegexp;
+    switch (compilerName) {
+      case "cl":
+        // cl does not have a version switch but it outputs the compiler version on stderr
+        // when no source files arguments are given
+        versionRegexp = /.* Compiler Version (.*) for .*/mg;
+        break;
+      case "gcc":
+      case "g++:":
+      case "cpp:":
+      case "c++:":
+        versionSwitch = "--version";
+        versionRegexp = RegExp(`${compilerName} .* (.*)`, "mg");
+        break;
+      case "clang":
+      case "clang-cl":
+      case "clang++":
+        versionSwitch = "--version";
+        versionRegexp = /clang version (.*) /mg;
+        break;
+      default:
+        return; // don't disclose information for compilers not in the known list
+    }
+
+    return `${compilerName}(${await this.testCompilerVersion(compilerName, compilerDir, versionSwitch, versionRegexp)})`;
```

Comment:
Can you please make the compiler name and compiler version be separate properties instead of putting them together in the same string?